File: src/test/java/testingil/webinar/tddapi/s06/new_unit_test/NavigatorUnitTests.java
package testingil.webinar.tddapi.s06.new_unit_test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import testingil.webinar.tddapi.s06.Distance;
import testingil.webinar.tddapi.s06.Location;
import testingil.webinar.tddapi.s06.Navigator;

import static org.hamcrest.MatcherAssert.assertThat;

class NavigatorUnitTests {

	@Test
	void when_created_distance_is_always_zero() {
		Location loc = new Location("London");
		Navigator nav = new Navigator(loc);
		Distance distance = nav.getDistanceFromDestination();
		assertThat(distance.inKm(), is(0));
	}

}

File: Python/s06_new_unit_test/navigator_unit_tests.py
import unittest

from hamcrest import assert_that, raises, calling, equal_to

from s06_new_unit_test.location import Location
from s06_new_unit_test.navigator import Navigator


class NavigatorUnitTests(unittest.TestCase):
    def test_when_created_distance_is_always_zero(self):
        loc = Location("London")
        nav = Navigator(loc)
        distance = nav.getDistanceFromDestination()
        assert_that(distance.inKm(), equal_to(0))


if __name__ == '__main__':
    unittest.main()
